### ISSUE TITLE
Open With not working for scripts

### ISSUE THREAD

**Issue by thany** [Microsoft/BashOnWindows#1078]
When I "open with" a shell script with Bash, I get this error message:
```
/bin/bash: D:\Path\To\Scripts\dostuff.sh: No such file or directory
```
Which is understandable, because Linux uses a different way to indicate paths.

**Expected results**

I'd expect it (not sure what "it" would be) to convert the supplied path to a path that is usable for bash, e.g.
`D:\Path\To\Scripts\dostuff.sh` -> `/mnt/d/Path/To/Scripts/dostuff.sh`

**Steps to repro**

1. Make sure you've got a shell script (a `.sh` file). Any will do.
2. Right click -> Open With -> Choose another app
3. More apps -> scroll down -> Look for another app on this pc
4. Type C:\Windows\system32\bash.exe (or whereever your Windows install is, you know the drill)

Now Bash will open very briefly with the above error message. Steps 2 to 4 can now be replaced with a simple doubleclick on a shell script, still resulting in the same error message.

Windows 10.0.14393 x64

**Comment by fpqc** [comment]
You need to do bash.exe -c, I believe.

**Comment by aseering** [comment]
@fpqc -- I believe that's true but I don't think it's sufficient?  Don't you also need to pass in a valid Linux path?

**Comment by fpqc** [comment]
@aseering bash.exe does automagic path translation iirc.

**Comment by benhillis** [comment]
@fpqc Relative paths should work. We do not translate absolute paths in bash.exe

**Comment by fpqc** [comment]
@benhillis yeah, that was my experience when I tested this out to get it working for a stackexchange question.

**Comment by rodrymbo** [comment]
I tried playing with this the other day. I've [made a VBS script](http://wsl-forum.qztc.io/viewtopic.php?f=6&t=42) that can act as a drop-target from Explorer for a bash script, though it would be quite nice if the bash.exe program itself would recognize a Windows path (beyond relative) and convert it.

I tried various ways to drop a script onto a shortcut in Explorer (for `bash.exe -c "somescript"`), to pass the dropped path along to "somescript" to process; but I couldn't find a way to do it. I think bash.exe is only allowing a certain number of arguments through into the WSL environment, at least from dropping onto a shortcut. I believe if the argument is inside the quotes ( `-c "somescript argument"`) it is passed along, but when dropping onto a shortcut it would end up appended after the -c quotes, bash.exe doesn't pass it in, and "somescript" never gets the argument. Maybe a Windows guru can tell me how to make a desktop shortcut that will pass the argument?

(Does that make sense?)

Anyway, the use case of making a shortcut to run one specific script in bash.exe is straightforward: figure out what the linux path would be, and use it with `-c`.  But the use case for dropping a script onto a shortcut (or using it with `Send to`, or even `Open`) does not seem to be possible at this point.

**Comment by thany** [comment]
@benhillis I'm sure relative paths work, but that's just not how "open with" works in Windows. It *always* supplies a full path.

No matter how this issue is twisted, in the end *something* needs to translate a Windows-path to a Linux-path.

So! I made a crude script that does exactly that. Like the VBscript thing @rodrymbo made, [I made a CMD script](https://github.com/thany/bash-assoc), which is about 100% less likely to set off virus alerts :)

**Comment by benhillis** [comment]
@thany I understand that.  Currently bash.exe takes its command line arguments and passes them to /bin/bash as-is (with the exception of the ~ argument which must be the first argument).  I think supporting Open-Wish (perhaps even registering bash.exe as a "handler" for .sh files) is a good suggestion and can look into what changes we'd have to make to support that.

**Comment by NINI1988** [comment]
For Everyone who has the same problem, I found a way with regedit instead of a script.

Change (Standart) in
HKEY_CLASSES_ROOT\Applications\bash.exe\shell\open\command\
to
"C:\Windows\System32\bash.exe" -c "$(wslpath '%1')"

Then select .sh file in explorer an choose open with: "C:\Windows\System32\bash.exe"

**Comment by thany** [comment]
Now all that Microsoft needs to do is add this as a default context menu action for `.sh` files if WSL is installed.
